Gems/Atom/RHI/Code/Source/RHI.Edit/ShaderBuildArguments.cpp

```diff
@@ -0,0 +1,162 @@
+                    ->Field("metallib", &ShaderBuildArguments::m_metalLibArguments)
+                    ;
+            }
+        }
+
+        ShaderBuildArguments::ShaderBuildArguments(
+                bool generateDebugInfo,
+                const AZStd::vector<AZStd::string>& preprocessorArguments,
+                const AZStd::vector<AZStd::string>& azslcArguments,
+                const AZStd::vector<AZStd::string>& dxcArguments,
+                const AZStd::vector<AZStd::string>& spirvCrossArguments,
+                const AZStd::vector<AZStd::string>& metalAirArguments,
+                const AZStd::vector<AZStd::string>& metalLibArguments)
+            : m_generateDebugInfo(generateDebugInfo)
+            , m_preprocessorArguments(preprocessorArguments)
+            , m_azslcArguments(azslcArguments)
+            , m_dxcArguments(dxcArguments)
+            , m_spirvCrossArguments(spirvCrossArguments)
+            , m_metalAirArguments(metalAirArguments)
+            , m_metalLibArguments(metalLibArguments)
+        {
+        }
+
+        bool ShaderBuildArguments::HasArgument(const AZStd::vector<AZStd::string>& argList, const AZStd::string& arg)
+        {
+            return AZStd::find(argList.begin(), argList.end(), arg) != argList.end();
+        }
+
+        void ShaderBuildArguments::AppendArguments(AZStd::vector<AZStd::string>& out, const AZStd::vector<AZStd::string>& in)
+        {
+            for (const auto& argument : in)
+            {
+                if (!HasArgument(out, argument))
+                {
+                    out.push_back(argument);
+                }
+            }
+        }
+
+        void ShaderBuildArguments::RemoveArguments(AZStd::vector<AZStd::string>& out, const AZStd::vector<AZStd::string>& in)
+        {
+            out.erase(remove_if(out.begin(), out.end(),
+                [&](const AZStd::string& arg)
+                {
+                    return HasArgument(in, arg);
+                }), std::end(out));
+        }
+
+        AZStd::string ShaderBuildArguments::ListAsString(const AZStd::vector<AZStd::string>& argList)
+        {
+            AZStd::string listAsString;
+            AZ::StringFunc::Join(listAsString, argList, " ");
+            return listAsString;
+        }
+
+        int ShaderBuildArguments::AppendDefinitions(const AZStd::vector<AZStd::string>& definitions)
+        {
+            int oldPreprocessorArgCount = aznumeric_caster(m_preprocessorArguments.size());
+
+            // Roll the Definitions into m_preprocessorArguments.
+            AZStd::vector<AZStd::string> definitionsAsArguments;
+            for (auto definition : definitions)
+            {
+                AZ::StringFunc::TrimWhiteSpace(definition, true, true);
+                if (definition.empty())
+                {
+                    AZ_Warning("ShaderBuildArguments", false, "%s Found an empty definition", __FUNCTION__);
```

Comment:
Print \_\_FUNCTION\_\_ isn't necessary, the AZ Trace system will provide this info automatically.